docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1135 @@
+1. Executing command.
+    1. `ProfileSession` calls `ProfileDelete.execute()` with the rest of parsed input.
+    1. `ProfileDelete` calls `ProfileStorage.loadData()` to load existing profile in the system. If there is no existing profile, `ProfileDelete` returns a failure result to `ProfileSession`. Otherwise, the process continues with step `iii`.
+    1. `ProfileDelete` calls `Ui.CheckConfirmation()` to get user's confirmation on the deletion since this action is irrevocable. If user  fails to confirm, `ProfileDelete` returns an abort result to `ProfileSession`. Otherwise, the process continues with step `iv`.
+    1. `ProfileDelete` calls `ProfileStorage.saveData()` to save a `null` object which represents a deleted profile.
+    1. `ProfileDelete` returns a result to `ProfileSession`.   
+1. Prompting result to user.
+    1. `ProfileSession` calls `CommandResult.getFeedbackMessage()` to get the execution feedback message.
+    1. `ProfileSession` calls `Ui.showToUser()` to show result to the user.
+
+All descriptions, warnings and responses will be handled by `Ui` to ensure consistence across the app.
+
+The sequence diagram below summarizes how deleting an added profile works:
+
+![Load Data Sequence Diagram](pictures/khoa/DeleteProfile.png)
+
+**Design considerations**
+
+Aspects: Loading of stored data
+
+- Alternative 1 (current choice): call public methods of Storage class to load the profile from hard disk every time the user wants to delete profile.
+
+|     |     |
+|-----|-----|
+|**Pros** | Profile data is up-to-date if the user prefers to edit it in text file rather than using commands in The Schwarzenegger.|  
+|**Cons** | Execution time is slow down due to loading the data.|
+
+- Alternative 2: call public methods of Storage class to load the profile from hard disk when user enter Profile Menu.
+
+|     |     |
+|-----|-----|
+|**Pros** | Execution time is fast.|  
+|**Cons** | Profile data is not updated in real time if user edits it in text file while running The Schwarzenegger.|
+
+### 4.3. Diet-related Features
+#### 4.3.1. Listing out all commands: `help`
+This command lists out all help commands in a typed list that indicates to the user all the commands available and how to use them.
+
+**Implementation**  
+When the user types `help` in a Diet Manager instance, the following sequence occurs. 
+1. The user keys in `help`.
+    
+    1. A `DietSessionUi` component will call `dietSessionUI.getInput()`. 
+    1. Input will be parsed in `processCommand()`.   
+    
+2. Creation of `DietSessionHelp` command object from input
+    1. This will create a `DietSessionHelp()` instantiation of which the method `execute()` is called.
+    
+3. Executing Command
+    1. The execute() method will call print out the list of commands onto the console with printHelpFormatter() from static CommonUi.java.
+[&#8593; Return to Top](#developer-guide)
+
```

Comment:
Would it be better to add a simple sequence diagram here to be consistent with the format?